test/lib/ufe/testGroupCmd.py

```diff
@@ -35,6 +34,38 @@
 import os
 import unittest
 
+def createStage():
+    ''' create a simple stage '''
+    cmds.file(new=True, force=True)
+    import mayaUsd_createStageWithNewLayer
+    proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+    proxyShapes = cmds.ls(type="mayaUsdProxyShapeBase", long=True)
+    proxyShapePath = proxyShapes[0]
+    proxyShapeItem = ufe.Hierarchy.createItem(ufe.PathString.path(proxyShapePath))
+    proxyShapeContextOps = ufe.ContextOps.contextOps(proxyShapeItem)
+    stage = mayaUsd.lib.GetPrim(proxyShapePath).GetStage()
+    return (stage, proxyShapePathStr, proxyShapeItem, proxyShapeContextOps)
+
+class SphereGenerator():
```

Comment:
Not too sure on the advantage of using yield versus calling a plain function.

Reply:
Simply the only advantage of "yield" here is that it saves the states ( in this case the "increment" ) and later continues from there on successive calls. 

I do see your point Pierre. This generator is super simple and could be designed a lot lot better but I didn't want to spend a lot of time on it. As you suggested, I could also go with plain old function.

Reply:
At this point it's doing the job, not enough value in spending time on this.